qiskit/pulse/library/symbolic_pulses.py

```diff
@@ -614,42 +607,67 @@ class Gaussian(metaclass=_PulseType):
     .. math::
 
         f'(x) &= \exp\Bigl( -\frac12 \frac{{(x - \text{duration}/2)}^2}{\text{sigma}^2} \Bigr)\\
-        f(x) &= \text{amp} \times \frac{f'(x) - f'(-1)}{1-f'(-1)}, \quad 0 \le x < \text{duration}
+        f(x) &= \text{A} \times  \frac{f'(x) - f'(-1)}{1-f'(-1)}, \quad 0 \le x < \text{duration}
 
-    where :math:`f'(x)` is the gaussian waveform without lifting or amplitude scaling.
+    where :math:`f'(x)` is the gaussian waveform without lifting or amplitude scaling, and
+    :math:`\text{A} = \text{amp} \times \exp\left(i\times\text{angle}\right)`.
     """
 
     alias = "Gaussian"
 
     def __new__(
         cls,
         duration: Union[int, ParameterExpression],
-        amp: Union[complex, ParameterExpression],
+        amp: Union[complex, float, ParameterExpression],
         sigma: Union[float, ParameterExpression],
+        angle: Optional[Union[float, ParameterExpression]] = None,
         name: Optional[str] = None,
         limit_amplitude: Optional[bool] = None,
     ) -> SymbolicPulse:
         """Create new pulse instance.
 
         Args:
             duration: Pulse length in terms of the sampling period `dt`.
-            amp: The amplitude of the Gaussian envelope.
+            amp: The magnitude of the amplitude of the Gaussian envelope.
+                    Complex amp support will be deprecated.
             sigma: A measure of how wide or narrow the Gaussian peak is; described mathematically
                    in the class docstring.
+            angle: The angle of the complex amplitude of the Gaussian envelope. Default value 0.
             name: Display name for this pulse envelope.
             limit_amplitude: If ``True``, then limit the amplitude of the
                 waveform to 1. The default is ``True`` and the amplitude is constrained to 1.
 
         Returns:
             SymbolicPulse instance.
+
+        Raises:
+            PulseError: If both complex amp and angle are provided as arguments.
         """
-        parameters = {"amp": amp, "sigma": sigma}
+        # This should be removed once complex amp support is deprecated.
+        if isinstance(amp, complex):
+            if angle is None:
+                warnings.warn(
+                    "Complex amp will be deprecated. "
+                    "Use float amp (for the magnitude) and float angle instead.",
+                    PendingDeprecationWarning,
```

Comment:
What is the plan for the deprecation here? I think it will be difficult to deprecate complex amplitude since that is what the backend reports for default pulse calibrations and so far we have not planned to change what the backend reports.

Reply:
@wshanks 
My assumption was that the backend might provide pulses with complex amp, but that it will always do so via qpy file. For this reason, qpy load now verifies the type of amp, and breaks into (amp,angle) when complex. Because of this, following the deprecation period, we shouldn't expect complex amp when creating a pulse. Do you think this is a suitable solution?

Reply:
This has been already addressed by this PR. The qobj converter consists of two classes, namely, `InstructionToQobjConverter` (client to server, i.e job data) and `QobjToInstructionConverter` (server to client, i.e. calibration data). You have updated both classes in this PR and calibration data is converted into amp, angle format [here](https://github.com/Qiskit/qiskit-terra/blob/8a74b70edbe60db99c3cfea35ac2090df0846a7b/qiskit/qobj/converters/pulse_instruction.py#L732-L735). So deprecation doesn't break calibration data.

Note that in the V2 IBM provider the payload is migrating to qpy but calibration data is still JSON format. In my PR https://github.com/Qiskit/qiskit-terra/pull/8885 I just implemented lazy conversion for pure-circuit users and the loading mechanism itself doesn't change.

Reply:
Okay, great. I just wanted to make sure loading the calibration data would not generate a warning.